File: tests/fixtures/go/simple.go
// FIXTURE: Simple Go functions
// TESTS: Basic function signature extraction

package main

import "fmt"

func Add(a int, b int) int {
    return a + b
}

func Greet(name string) string {
    return fmt.Sprintf("Hello, %s!", name)
}

type Calculator struct {
    value int
}

func (c *Calculator) Add(x int) int {
    return c.value + x
}

type Computer interface {
    Compute(x int) int
}

type Status int

const (
    Active Status = iota
    Inactive
    Pending
)

File: tests/fixtures/rust/simple.rs
//! FIXTURE: Simple Rust functions
//! TESTS: Basic function signature extraction

pub fn add(a: i32, b: i32) -> i32 {
    a + b
}

pub fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}

pub struct Calculator {
    value: i32,
}

impl Calculator {
    pub fn new(value: i32) -> Self {
        Self { value }
    }

    pub fn add(&self, x: i32) -> i32 {
        self.value + x
    }
}

pub trait Compute {
    fn compute(&self, x: i32) -> i32;
}

pub enum Status {
    Active,
    Inactive,
    Pending,
}
